Constructor type error in 5.2.0

<!-- Please don't delete this template otherwise your issue will be closed immediately -->
<!-- Before creating an issue please make sure you are using the latest version of webpack. -->

### Bug report

<!-- Please ask questions on discussions, StackOverflow or the webpack Gitter. -->
<!-- https://github.com/webpack/webpack/discussions -->
<!-- https://stackoverflow.com/questions/ask?tags=webpack -->
<!-- https://gitter.im/webpack/webpack -->
<!-- Issues which contain questions or support requests will be closed. -->

### Actual Behavior

Using the plugin throws a typescript type error as of version 5.2.0:
```
error TS2351: This expression is not constructable.
  Type 'typeof import("/Users/user/app/node_modules/terser-webpack-plugin/types/index")' has no construct signatures.

259       minimizer: [new TerserPlugin()]
```

<!-- Explain exactly how it behave -->

### Expected Behavior

Does not throw an error for documented usage

### How Do We Reproduce?

Simply using it as documented
```js
// @ts-check
const TerserPlugin = require('terser-webpack-plugin')

module.exports = {
  optimization: {
    minimize: true,
    minimizer: [new TerserPlugin()]
  }
}
```

--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
     "url": "https://opencollective.com/webpack"
   },
   "main": "dist/cjs.js",
-  "types": "types/index.d.ts",
+  "types": "types/cjs.d.ts",
   "engines": {
     "node": ">= 10.13.0"
   },

